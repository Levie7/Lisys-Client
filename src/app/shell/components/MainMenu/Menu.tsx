import { Menu as AntMenu } from 'antd';
import * as React from 'react';

import { Menu as MenuAPI } from 'src/core/api';

import { Icon } from 'src/shared/components/Icon';
import { Link } from 'src/shared/components/Link';

import { menuData } from './constants';

interface MenuProps {
    className?: string;
    mode?: 'vertical' | 'vertical-left' | 'vertical-right' | 'horizontal' | 'inline';
    theme?: 'light' | 'dark';
}

export const Menu = React.memo<MenuProps>(({ className, mode, theme }) => {
    function generateMenu(menuItems: any) {
        return menuItems
            .filter((menu: MenuAPI) => menu.status === 'Active')
            .map((menu: MenuAPI) => (menu.children ? renderSubMenu(menu) : renderMenuItem(menu)));
    }

    function renderMenuItem(menu: MenuAPI) {
        return (
            <AntMenu.Item key={menu.key}>
                <Link to={menu.url}>
                    {Icon[menu.icon]}
                    <span>{menu.name}</span>
                </Link>
            </AntMenu.Item>
        );
    }

    function renderSubMenu(menu: MenuAPI) {
        let subMenuTitle = (
            <>
                {Icon[menu.icon]}
                <span>{menu.name}</span>
            </>
        );

        return (
            <AntMenu.SubMenu key={menu.key} title={subMenuTitle}>
                {generateMenu(menu.children)}
            </AntMenu.SubMenu>
        );
    }

    return (
        <AntMenu className={className} mode={mode} theme={theme}>
            {generateMenu(menuData)}
        </AntMenu>
    );
});
