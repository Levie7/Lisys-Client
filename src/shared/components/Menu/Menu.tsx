import { Menu as AntMenu } from 'antd';
import { ClickParam } from 'antd/lib/menu';
import MenuItem from 'antd/lib/menu/MenuItem';
import * as React from 'react';

interface MenuProps {
    className?: string;

    onClick?: (param: ClickParam) => void;
}

export class Menu extends React.PureComponent<MenuProps> {
    static Item: typeof MenuItem;

    render() {
        let { children, className, onClick, ...props } = this.props;

        return (
            <AntMenu {...props} className={className} onClick={onClick}>
                {children}
            </AntMenu>
        );
    }
}
Menu.Item = AntMenu.Item;
