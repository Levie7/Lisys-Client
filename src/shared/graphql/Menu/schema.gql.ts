import { Menu } from 'src/core/api';
import { gql, useQuery } from 'src/core/graphql';

export const MENUS = gql`
    query {
        getMenus {
            children {
                icon
                id
                key
                name
                status
                url
            }
            icon
            id
            key
            name
            status
            url
        }
    }
`;

export const getMenus = (options: any) => useQuery<{ getMenus: Menu[] }>(MENUS, options);
