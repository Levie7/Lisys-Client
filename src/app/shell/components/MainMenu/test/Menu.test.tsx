import { shallow } from 'enzyme';
import React from 'react';

import { Menu } from '../Menu';

describe('Menu', () => {
    const wrap = shallow(<Menu />);

    it('<Menu/> - should render Menu component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should contains all Menu children', () => {
        expect(wrap.find('Menu').exists()).toBeTruthy();
        expect(wrap.find('MenuItem').exists()).toBeTruthy();
        expect(wrap.find('SubMenu').exists()).toBeTruthy();
    });
});
