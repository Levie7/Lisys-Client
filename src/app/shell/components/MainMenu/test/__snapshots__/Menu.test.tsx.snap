// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu <Menu/> - should render Menu component correctly 1`] = `ShallowWrapper {}`;
